### PR TITLE
fix masking dimension mismatch

### PR DESCRIPTION
**Pull request by Data-Iab**
Fix :wrench: : AugmentedTensor can be masked without any need to expand the mask dimensions.

### CODE REVIEW

**Comment by Data-Iab** [comment]
Potential fix for this [issue](https://github.com/Visual-Behavior/aloception/issues/160)

File: aloscene/tensors/augmented_tensor.py [hunk 1]
```diff
@@ -302,7 +302,9 @@ def __getitem__(self, idx):
                 )
 
         if isinstance(idx, torch.Tensor):
-            tensor = torch.Tensor.__getitem__(self.rename(None), idx).reset_names()
+            if not all(e in [0, 1] for e in idx.unique().int()):
```
Comment (thibo73800) [inline aloscene/tensors/augmented_tensor.py:305]: I'm just worry about this line since it make filtering data slower by getting the `unique()` of the `idx`. What what the purpose of raising an Exception here ? Just getting a better error message ? If so, you can try catch the tensor & then check the possible source of error.

**Review by thibo73800 — CHANGES_REQUESTED** [review]

File: aloscene/tensors/augmented_tensor.py (hunk 1 above)
Comment (Data-Iab) [reply aloscene/tensors/augmented_tensor.py:305]: @thibo73800 It is very important to raise the exception to make sure the user is not unintentionally passing something else rather than a filter. For the time problem you're right, here is a potential solution that can replace this line:
`if not torch.equal(idx ** 3, idx)` as the only solution for `x ** 3 = x` equation is either 0 or 1, it ensures that they are the only values `idx` have.
Concerning running time, you can take a look at the results of the following code:
```python

m = torch.ones((1, 500, 500, 3))

t0 = time.time()
for _ in range(1000):
  if not torch.equal(m ** 3, m):
    pass


t1 = time.time()
print("runtime with torch.equal", t1 - t0)

t0 = time.time()
for _ in range(1000):
  if not all(e in [0, 1] for e in m.unique().int()):
    pass


t1 = time.time()
print("runtime with with previous solution", t1 - t0)
```
```
runtime with torch.equal 1.6653542518615723
runtime with with previous solution 17.1923611164093
```

Comment (thibo73800) [reply aloscene/tensors/augmented_tensor.py]: Ok, what happen if `the user is unintentionally passing something else rather than a filter. ` ? The code will fail ?

**Comment by thibo73800** [comment]
What is the status of this PR ?

**Review by thibo73800 — APPROVED** [review]
